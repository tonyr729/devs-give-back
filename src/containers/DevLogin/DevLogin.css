div {
  background-color: #03a9f4;
  width: 100vw;
  height: 100vh; }

.dev-login-message {
  font-size: 5rem;
  padding-top: 12rem;
  color: #ffffff; }

.login-button-github {
  border: none;
  width: 12rem;
  height: 2.9rem;
  background-repeat: no-repeat;
  background-image: url("../../images/github_icon.svg");
  background-size: 3rem;
  background-position-x: .3rem;
  background-color: #000;
  border-radius: .5rem;
  padding-left: 3rem;
  font-size: 1rem;
  color: #f8f8ff;
  margin: 4rem auto;
  display: block; }
  .login-button-github:hover {
    background-color: #06b900; }
  .login-button-github:active {
    color: #f8f8ff;
    background-color: #007e11; }
  .login-button-github:focus {
    outline: none; }
