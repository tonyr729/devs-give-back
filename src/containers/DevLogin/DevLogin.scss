div {
  background-color: #03a9f4;
  width: 100vw;
  height: 100vh;
}

.login-message {
  font-size: 5rem;
  padding-top: 12rem;
}

.login-button-github {
  border: none;
  width: 12rem;
  height: 2.9rem;
  background-repeat: no-repeat;
  background-image: url(/static/media/github_icon.5e2d57ae.svg);
  background-size: 3rem;
  background-position-x: .3rem;
  background-color: #000;
  border-radius: .5rem;
  padding-left: 3rem;
  font-size: 1rem;
  color: #f8f8ff;
  margin: 4rem auto;
  display: block;
  &:hover {
    background-color: #06b900;
  }
  &:active {
    color: #f8f8ff;
    background-color: #007e11;
  }
  &:focus {
    outline: none;
  }
}