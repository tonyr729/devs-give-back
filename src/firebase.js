import firebase from 'firebase';

const config = {
  apiKey: "AIzaSyAVk4m35HpmPoi0Rr1K495lVyfskmVDiD4",
  authDomain: "devsgiveback.firebaseapp.com",
  databaseURL: "https://devsgiveback.firebaseio.com",
  projectId: "devsgiveback",
  storageBucket: "devsgiveback.appspot.com",
  messagingSenderId: "171321681886"
};
firebase.initializeApp(config);
export default firebase;