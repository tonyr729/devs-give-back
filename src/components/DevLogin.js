import React, { Component } from 'react';
import firebase from '../firebase';

const DevLogin = () => {

  const gitHubLogin = async () => {
    try {
      const provider = new firebase.auth.GithubAuthProvider().addScope('user')
      const response = await firebase.auth().signInWithPopup(provider);
      const user = response.user;
      const repo = response.repo;
      console.log(response)
    } catch(error) {
      const errorCode = error.code;
      const errorMessage = error.message;
      const email = error.email;
      const credential = error.credential;
      const errorInfo = { errorCode, errorMessage, email, credential};
      console.log(errorInfo)
    };
  };

  return (
    <div>
      <button onClick={()=> gitHubLogin()}>Login with GitHub</button>
    </div>
  );
}


export default DevLogin;