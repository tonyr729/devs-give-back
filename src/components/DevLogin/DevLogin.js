import React, { Component } from 'react';
import firebase from '../../firebase/firebase';

export const DevLogin = () => {

  const gitHubLogin = async () => {
    try {
      const provider = new firebase.auth.GithubAuthProvider()
      const response = await firebase.auth().signInWithPopup(provider);
      const user = response.user;
      const repo = response.repo;
    } catch(error) {
      const errorCode = error.code;
      const errorMessage = error.message;
      const email = error.email;
      const credential = error.credential;
      const errorInfo = { errorCode, errorMessage, email, credential};
    };
  };

  return (
    <div>
      <button onClick={()=> gitHubLogin()}>Login with GitHub</button>
    </div>
  );
}

