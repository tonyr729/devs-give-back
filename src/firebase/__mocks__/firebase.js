import * as firebase from 'firebase';

const onAuthStateChanged = jest.fn();

const getRedirectResult = jest.fn(() => {
  return Promise.resolve({
    user: {
      displayName: 'redirectResultTestDisplayName',
      email: 'redirectTest@test.com',
      emailVerified: true
    }
  })
})

const sendEmailVerification = jest.fn(() => {
  return Promise.resolve('result of sendEmailVerification');
})

const sendPasswordResetEmail = jest.fn(() => Promise.resolve());

const createUserWithEmailAndPassword = jest.fn(() => {
  return Promise.resolve('result of createUserWithEmailAndPassword');
})

const signInWithEmailAndPassword = jest.fn(() => {
  return Promise.resolve('result of signInWithEmailAndPassword');
})

const signInWithPopup = jest.fn(() => {
  return Promise.resolve('result of signInWithPopup');
})

const initializeApp = jest
  .spyOn(firebase, 'initializeApp')
  .mockImplementation(() => {
    return {
      auth: () => {
        return {
          createUserWithEmailAndPassword,
          signInWithEmailAndPassword,
          currentUser: {
            sendEmailVerification
          }
        }
      }
    }
  })

jest.spyOn(firebase, 'auth').mockImplementation(() => {
  return {
    onAuthStateChanged,
    currentUser: {
      displayName: 'testDisplayName',
      email: 'test@test.com',
      emailVerified: true
    },
    signInWithPopup
  };
});

firebase.auth.GithubAuthProvider = jest.fn(() => {})
firebase.auth.GoogleAuthProvider = jest.fn(() => {})

export default firebase;