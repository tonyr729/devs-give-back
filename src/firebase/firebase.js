import firebase from 'firebase';
import { firebaseKey } from './apiKeys';

const config = {
  apiKey: firebaseKey,
  authDomain: "devsgiveback.firebaseapp.com",
  databaseURL: "https://devsgiveback.firebaseio.com",
  projectId: "devsgiveback",
  storageBucket: "devsgiveback.appspot.com",
  messagingSenderId: "171321681886"
};
firebase.initializeApp(config);

export default firebase;