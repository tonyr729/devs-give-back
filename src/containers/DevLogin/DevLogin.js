import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import { signInDev, devError} from '../../actions/actions';
import DataCleaner from '../../helpers/DataCleaner';
import DatabaseHelper from '../../helpers/DatabaseHelper';
import './DevLogin.css';

export class DevLogin extends Component {
  constructor() {
    super();
    this.database = new DatabaseHelper();
    this.cleaner = new DataCleaner();
  }


  handleLogin = async () => {
    try {
      const result = await this.database.gitHubLogin();
      const cleanDev = this.cleaner.cleanDevLogin(result.user, result.token);
      this.props.signInDev(cleanDev);
      this.database.writeDevToDatabase(cleanDev);
    } catch (error) {
      const cleanError = this.cleaner.cleanError(error);
      this.props.devError(cleanError);
      this.props.history.push("/error-page");
    }
  };

  logInCheck = (dev) => {
    const value = Object.keys(dev).length;
    if ( value !== 0 ) {
      return (<Redirect to='/dev-profile'/>);
    }
  }

  render() {
    const redirect = this.logInCheck(this.props.dev);
    return (
      <div>
        { redirect }
        <p className='login-message'>Please login</p>
        <button className='login-button-github'onClick={()=> this.handleLogin()}>Login with GitHub</button>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  dev: state.dev
});

export const mapDispatchToProps = (dispatch) => ({
  signInDev: (user, token) => dispatch(signInDev(user, token)),
  devError: (error) => dispatch(devError(error))
});

export default connect(mapStateToProps, mapDispatchToProps)(DevLogin);

