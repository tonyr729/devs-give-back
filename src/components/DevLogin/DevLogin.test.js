import { shallow } from 'enzyme';
import React from 'react';
import { DevLogin } from './DevLogin';
import firebase from '../../firebase/firebase';
jest.mock('../../firebase/firebase');

describe('DevLogin', () => {
  it('should match snapshot', () => {
    const devLogin = shallow(<DevLogin  />)

    expect(devLogin).toMatchSnapshot();
  });
  
  describe('gitHubLogin', () => {
    let devLogin;

    beforeEach(() => {
      devLogin = shallow(<DevLogin />);
    });
  
    it('should call the expected methods of firebase', () => {
      devLogin.find('button').simulate('click');

  
      expect(firebase.auth.GithubAuthProvider).toHaveBeenCalled();
      expect(firebase.auth().signInWithPopup).toHaveBeenCalled();
    });
  });
});


  
