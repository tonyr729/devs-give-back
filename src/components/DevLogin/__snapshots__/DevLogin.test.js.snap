// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevLogin should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        onClick={[Function]}
>
        Login with GitHub
</button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Login with GitHub",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Login with GitHub",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          onClick={[Function]}
>
          Login with GitHub
</button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login with GitHub",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login with GitHub",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
